### PR TITLE
Make compatible with jax v0.3.5

### PR DESCRIPTION
**Pull request by Michael-T-McCann**
The main problem was that `x.copy()` no longer turns a `DeviceArray` into an `ndarray`. Instead, use `np.array(x)` or `x.from_device()`.

### CODE REVIEW

File: scico/loss.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 from functools import wraps
 from typing import Callable, Optional, Union
 
+import jax
```
Comment (bwohlberg) [inline scico/loss.py:16]: No other changes to this module, but now `jax` is a required import? That's rather strange.

**Review by bwohlberg — APPROVED** [review]

File: scico/loss.py (hunk 1 above)
Comment (Michael-T-McCann) [reply scico/loss.py:16]: Yup, it's a code smell that it was never a problem before. The important line:

```python
if snp.isscalar(other) or isinstance(other, jax.core.Tracer):
```
